File: docs/breez-sdk/snippets/swift/BreezSdkSnippets/Sources/ListPayments.swift
import BreezSdkSpark
import Foundation

func getPayment(sdk: BreezSdk) async throws -> Payment {
    // ANCHOR: get-payment
    let paymentId = "<payment id>"
    let response = try await sdk.getPayment(
        request: GetPaymentRequest(paymentId: paymentId)
    )
    let payment = response.payment
    // ANCHOR_END: get-payment
    return payment
}

func listPayments(sdk: BreezSdk) async throws -> [Payment] {
    // ANCHOR: list-payments
    let response = try await sdk.listPayments(
        request: ListPaymentsRequest())
    let payments = response.payments
    // ANCHOR_END: list-payments
    return payments
}

func listPaymentsFiltered(sdk: BreezSdk) async throws -> [Payment] {
    // ANCHOR: list-payments-filtered
    // Filter by asset (Bitcoin or Token)
    let assetFilter = AssetFilter.token(tokenIdentifier: "token_identifier_here")
    // To filter by Bitcoin instead:
    // let assetFilter = AssetFilter.bitcoin

    let response = try await sdk.listPayments(
        request: ListPaymentsRequest(
            // Filter by payment type
            typeFilter: [PaymentType.send, PaymentType.receive],
            // Filter by status
            statusFilter: [PaymentStatus.completed],
            assetFilter: assetFilter,
            // Time range filters
            fromTimestamp: 1_704_067_200,  // Unix timestamp
            toTimestamp: 1_735_689_600,  // Unix timestamp
            // Pagination
            offset: 0,
            limit: 50,
            // Sort order (true = oldest first, false = newest first)
            sortAscending: false
        ))
    let payments = response.payments
    // ANCHOR_END: list-payments-filtered
    return payments
}

File: docs/breez-sdk/snippets/flutter/lib/list_payments.dart
import 'package:breez_sdk_spark_flutter/breez_sdk_spark.dart';

Future<Payment> getPayment(BreezSdk sdk) async {
  // ANCHOR: get-payment
  String paymentId = "<payment id>";
  GetPaymentRequest request = GetPaymentRequest(paymentId: paymentId);
  GetPaymentResponse response = await sdk.getPayment(request: request);
  Payment payment = response.payment;
  // ANCHOR_END: get-payment
  return payment;
}

Future<List<Payment>> listPayments(BreezSdk sdk) async {
  // ANCHOR: list-payments
  ListPaymentsRequest request = ListPaymentsRequest();
  ListPaymentsResponse response = await sdk.listPayments(request: request);
  List<Payment> payments = response.payments;
  // ANCHOR_END: list-payments
  return payments;
}

Future<List<Payment>> listPaymentsFiltered(BreezSdk sdk) async {
  // ANCHOR: list-payments-filtered
  // Filter by asset (Bitcoin or Token)
  AssetFilter assetFilter = AssetFilter.token(tokenIdentifier: "token_identifier_here");
  // To filter by Bitcoin instead:
  // AssetFilter assetFilter = AssetFilter.bitcoin();

  ListPaymentsRequest request = ListPaymentsRequest(
    // Filter by payment type
    typeFilter: [PaymentType.send, PaymentType.receive],
    // Filter by status
    statusFilter: [PaymentStatus.completed],
    assetFilter: assetFilter,
    // Time range filters
    fromTimestamp: BigInt.from(1704067200), // Unix timestamp
    toTimestamp: BigInt.from(1735689600),   // Unix timestamp
    // Pagination
    offset: 0,
    limit: 50,
    // Sort order (true = oldest first, false = newest first)
    sortAscending: false,
  );
  ListPaymentsResponse response = await sdk.listPayments(request: request);
  List<Payment> payments = response.payments;
  // ANCHOR_END: list-payments-filtered
  return payments;
}
